### ISSUE TITLE
Deep recursion problem

### ISSUE THREAD

**Issue by zensubz** [trizen/youtube-viewer#175]
Hi,

Gtk-YouTube-Viewer doesn't run for me, and the following is the output if I run it from terminal:

**Deep recursion on subroutine "WWW::YoutubeViewer::lwp_get" at /usr/lib/perl5/vendor_perl/5.22.3/WWW/YoutubeViewer.pm line 360.**

The youtube-viewer window just hangs and also the CPU usage is quite high during this process.

Kindly suggest a method to overcome this issue. I have all the dependencies installed. Thanks.

**Comment by trizen** [comment]
Hi. Please open the configuration file of `gtk-youtube-viewer` (`~/.config/youtube-viewer/gtk-youtube-viewer.conf`), set `debug` to `1`, then execute `gtk-youtube-viewer` inside a terminal and copy-paste the error messages here.

Most probably you have a YouTube server blocked (in `/etc/hosts`, maybe?) and `gtk-youtube-viewer` tries to recover from a failed connection to an URL by retrying to connect, which leads to a recursive process of failed attempts (which needs to be fixed).

If this is the case, you can fix the issue by unblocking the following two URLs: 

* http://i1.ytimg.com/ - used for thumbnails
* https://www.googleapis.com/ - YouTube API

**Comment by zensubz** [comment]
Hi,

Thanks for the quick response-cum-fix. I applied the same, and now gtk-youtube-viewer has come to life..! The hosts file didn't have the URL entries you have mentioned, but I had to install two more perl-modules (perl-Net-SSLeay and perl-Types-Serialiser), as I got a warning "...Crypt-SSLeay can't verify hostnames" ...

Deleted all the (old) config/cache, re-started the application, and now it works in all its glory. And no error messages using the debug=0 method you have mentioned.
